testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/SAMLServletAdapterTest.java

```diff
@@ -939,7 +939,7 @@ public void salesPostSigTest() {
     @Test
     // https://issues.jboss.org/browse/KEYCLOAK-3971
     public void salesPostSigTestUnicodeCharacters() {
-        final String username = "ěščřžýáíRoàåéèíñòøöùüßÅÄÖÜ";
+        final String username = "ěščřžýáíroàåéèíñòøöùüßåäöü";
```

Comment:
Why is this necessary?

Reply:
`MapUserProvider` doesn't use `username.toLowerCase()` when querying users. Therefore the test doesn't work when there are some upper case characters in username. 

This test is about testing username with special characters and username lettercase will be handled within https://github.com/keycloak/keycloak/issues/10245.

Another possibility would be to skip the test for now till https://github.com/keycloak/keycloak/issues/10245 is done, but I prefer this solution. wdyt?

Reply:
Is #10245 somehow relevant to querying users, isn't it only about stored lettercase? If we choose to store username preserving the lettercase, do we want to allow two users with usernames `admin` and `Admin`? I am asking because currently, I see that we query usernames with `ILIKE` twice and once with `EQ`, should we switch `EQ` to `ILIKE`? Would it solve the issue with this test?

Reply:
> I see that we query usernames with `ILIKE` twice and once with `EQ`, should we switch `EQ` to `ILIKE`?

There definitely needs to be a consistency in the search by username. If that is the case, could you please introduce an issue with direct references to the code to fix this? 

The other question is how each storage treats this `EQ` / `ILIKE` inside it. Some are case-sensitive and some case-insensitive. This is treated in #10245

Reply:
I created an issue here: https://github.com/keycloak/keycloak/issues/11602, feel free to add more details